tests/e2e/tests/checkout/saved-card.spec.js

```diff
@@ -0,0 +1,79 @@
+const { test, expect } = require( '@playwright/test' );
+import config from 'config';
+import { payments, api } from '../../utils';
+
+const {
+	emptyCart,
+	setupProductCheckout,
+	setupCheckout,
+	fillCardDetails,
+} = payments;
+const { CONSUMER_KEY, CONSUMER_SECRET } = process.env;
+
+let username, userEmail;
+
+test.beforeAll( async () => {
+	// This allow multiple tests to run in parallel.
+	const randomString = Date.now();
+	userEmail = randomString + '+' + config.get( 'users.customer.email' );
+	username = randomString + '.' + config.get( 'users.customer.username' );
+
+	const user = {
+		...config.get( 'users.customer' ),
+		...config.get( 'addresses.customer' ),
+		email: userEmail,
+		username,
+	};
+
+	await api.create.customer( user );
+} );
+
+test( 'customer can checkout with a saved card @smoke', async ( { page } ) => {
+	await test.step( 'customer login', async () => {
+		await page.goto( `/wp-admin` );
+		await page.fill( 'input[name="log"]', username );
+		await page.fill(
+			'input[name="pwd"]',
+			config.get( 'users.customer.password' )
+		);
+		await page.click( 'text=Log In' );
+	} );
+
+	await test.step( 'checkout and choose to save the card', async () => {
+		await emptyCart( page );
+
+		await setupProductCheckout( page );
+		await setupCheckout( page, null, true );
+		await fillCardDetails( page, config.get( 'cards.basic' ) );
+
+		// check box to save payment method.
+		await page.locator( '#wc-stripe-new-payment-method' ).click();
+
+		await page.locator( 'text=Place order' ).click();
+
+		await page.waitForNavigation();
+		await expect( page.locator( 'h1.entry-title' ) ).toHaveText(
+			'Order received'
+		);
+	} );
+
+	await test.step( 'checkout and pay with the saved card', async () => {
+		await emptyCart( page );
+		await setupProductCheckout( page );
+		await setupCheckout( page, null, true );
```

Comment:
It looks like these 3 steps are being used, and are gonna be needed for many future test cases.

I think it's fine as is, but we might consider moving it to a single function later.

Reply:
I'll keep that in mind for the next PR with test cases.